### PR TITLE
Consume API feature

### PR DESCRIPTION
**Pull request by zewdu444**
# Consuming  leaderboard API

- [x] To communicate with the REST API, Async/Await is used.
- [x] POST and GET method used
- [x] Submit and Refresh score were added
- [x] Readme file updated
- [x] Dist. files has been generated
- [x] Live deployed in GitHub

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,2 +1,35 @@
...
 import '../node_modules/@fortawesome/fontawesome-free/css/all.css';
+import sendScore from './module/sendScore.js';
+import displayScore from './module/displayScore.js';
+
+const scoreSubmit = document.getElementById('scoresubmit');
+const scoreUser = document.getElementById('user');
+const score = document.getElementById('score');
+const scoreRefresh = document.getElementById('refresh');
+// initial display score
+displayScore();
```
Comment (mckent05) [inline src/index.js]: **OPTIONAL** - To make your code more readable and easier to read, I suggest that you call this function when attached to an event listener, in this case, I suggest you use `DOMContentLoaded`. Please fix this 😄

File: src/index.js [hunk 2]
```diff
@@ -1,2 +1,35 @@
...
+      e.preventDefault();
+    }
+  });
+};
+// refresh score
+const refreshScore = async () => {
+  scoreRefresh.addEventListener('click', async () => {
+    window.location.reload();
+    displayScore();
+  });
```
Comment (mckent05) [inline src/index.js]: - According to the project requirements, `When the Refresh button is clicked, it should hit the GET API endpoint causing the scores list to change`.

File: src/index.js [hunk 3]
```diff
@@ -1,2 +1,35 @@
...
+};
+
+// call submit and refresh score
+submitScore();
+refreshScore();
```
Comment (mckent05) [inline src/index.js]: **OPTIONAL** - I do not advise calling these functions by default when your DOM loads, it doesn't help with the readability of your site. Please fix this to help improve your site's readability 🙏. The score should be submitted when the `Submit` button is clicked i.e hit the POST API
**OPTIONAL** - Also, the leaderboard should only be refreshed when the refresh button is clicked i.e hit the GET API, please fix this 😄

**Review by mckent05 — CHANGES_REQUESTED** [review]
Hi @zewdu444, 

# STATUS: CHANGES REQUIRED ♻️ ♻️ 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
almost there

## Highlights
- All linter checks are passing ✔️ ✔️ 
- Your PR is professional 👍 👍 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/index.js (hunk 1 above)
Comment (zewdu444) [reply src/index.js]: @mckent05  thanks for the feedback I fixed accordingly.

File: src/index.js (hunk 2 above)
Comment (zewdu444) [reply src/index.js]: @mckent05 thanks  i added a method which called GET API ( await getScore();)

File: src/index.js (hunk 3 above)
Comment (zewdu444) [reply src/index.js]: @mckent05 fixed both optional comment.

**Review by victorokoroji — APPROVED** [review]
Hi @zewdu444, 

# ✔️   Status: APPROVED 🏆 🎉 🎊 

You have done a great job on this project, keep it up 👍 🥇. Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
congratulations!  🎉💯🌟 

## Top highlight 🎯

- The Pull Request is following the Git Flow ✅ 
- The PR title and description are professional and descriptive ✅  
- Linters are working properly ✅ 



## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**